src/Compilers/CSharp/Portable/Binder/Binder.cs

```diff
@@ -788,11 +776,11 @@ internal Binder WithPatternVariablesIfAny(ArgumentListSyntax initializerArgument
             patterns.Free();
         }
 
-        internal BoundExpression WrapWithVariablesIfAny(BoundExpression expression)
+        internal BoundExpression WrapWithVariablesIfAny(CSharpSyntaxNode node, BoundExpression expression)
         {
             return (Locals.Length == 0)
                 ? expression
-                : new BoundSequence(expression.Syntax, Locals, ImmutableArray<BoundExpression>.Empty, expression, expression.Type) { WasCompilerGenerated = true };
+                : new BoundSequence(node, Locals, ImmutableArray<BoundExpression>.Empty, expression, expression.Type) { WasCompilerGenerated = true };
```

Comment:
It has the same impact as for any other compiler generated node. It is one of them and, therefore, should be marked as such. This is a new helper specific to patterns feature.